hvplot/converter.py

```diff
@@ -387,6 +396,42 @@ def __init__(self, data, x, y, kind=None, by=None, use_index=True,
             param.main.warning('Plotting {kind} plot with parameters x: {x}, '
                                'y: {y}, by: {by}, groupby: {groupby}'.format(**kwds))
 
+    def _process_clim(self, clim, center):
+        if clim is None:
+            if is_xarray(self.data):
+                data = self.data[list(self.data.data_vars)[0]]
+            else:
+                data = self.data
+
+            # round to the nearest 5, 0.5, 0.05, 0.005, 0.0005, etc
+            cmin = roundn(np.nanquantile(data, 0.05), method='down')
+            cmax = roundn(np.nanquantile(data, 0.95), method='up')
+            clim = (cmin, cmax)
+            auto = True
+        else:
+            cmin, cmax = clim
+            auto = False
+
+        if center is None:
+            divergent = cmin < 0 and cmax > 0
+            # if the order of magnitudes equate; we don't want to
+            # center the clim or use a divergent colormap
+            # if cmin == -1 and cmax == 20 since it's mostly positive
+            equal_oom = order_magn(cmin) == order_magn(cmax)
+            center = divergent and equal_oom
+
+        if center:
+            if auto:  # if user provided clim, don't change it
+                cmax = np.max([np.abs(cmin), np.abs(cmax)])
+                cmin = -cmax  # make top == bottom
+                clim = (cmin, cmax)
+
+            # if cmap isn't explicitly provided and the values center
+            # around 0 use the RdBu_r colormap instead of fire colormap
+            if not self._style_opts.get('cmap', None):
+                self._style_opts['cmap'] = 'RdBu_r'
```

Comment:
I don't think this colormap name should be hardcoded; for one thing it's not perceptually uniform.  The percetpually uniform `coolwarm` or `CET_D1A` palettes from [Colorcet](http://colorcet.pyviz.org/user_guide/Continuous.html) could be used, or `balance` from [cmocean](https://matplotlib.org/cmocean), but in any case I think it should be a parameter somewhere that the user can modify.

Reply:
It can be modified using specifying a cmap, unless you mean adding this functionality to holoviews' `bokeh/__init__.py` and `matplotlib/__init__.py` store?

Reply:
This PR provides the ability to automatically switch to a divergent colormap, right?  I'm saying that the colormap that is switched to in such a case should be configurable.  `coolwarm` from Matplotlib isn't perceptually uniform, while `coolwarm` from colorcet isn't necessarily available, so people aren't going to agree on what should be switched to.  So users should be able to define something like `hvplot.default_diverging_cmap="CET_D1A"`, allowing them to choose the diverging colormap without having to lose the auto-switching.  @philippjfr would be able to say where such a default could be defined.

Reply:
I'd prefer if we defined the default colormaps for these three cases in holoviews on ``hv.config``:

* numerical
* categorical
* diverging

Cc: @jlstevens

Reply:
I'd call the first one `sequential`, but otherwise sounds good!

Reply:
Seems reasonable to me. This just acknowledges the fact there are different types of colormaps so having multiple default makes sense.

Reply:
I'm unfamiliar with using hv.config; is hv.config under holoviews/holoviews/core/util.py/Config (class)

If so, do I just add a param?

Reply:
@jlstevens , can you advise?